src/engine/ruleset/decoders/nginx/nginx-access.yml

```diff
@@ -0,0 +1,174 @@
+    # 1.128.3.4 2a02:cf40:10ff:f00f:0000:0000:0:8000, 10.225.192.17 10.2.2.121 - - [30/Dec/2016:06:47:09 +0000] "GET /test.html HTTP/1.1" 404 8571 "-" "Mozilla/5.0 (compatible; Facebot 1.0; https://developers.facebook.com/docs/sharing/webmasters/crawler)"
+    - ~destination_address: >-
+         <destination.ip>(?:<destination.port>) <~ip1/ip>(?,) <~ip2/ip>(?,) <~ip3/ip>
+
+    # More than 3 Ips
+    # 1.128.3.4, 2a02:cf40:10ff:f00f:0000:0000:0:8000, 10.225.192.17 10.2.2.121 - - [30/Dec/2016:06:47:09 +0000] "GET /test.html HTTP/1.1" 404 8571 "-" "Mozilla/5.0 (compatible; Facebot 1.0; https://developers.facebook.com/docs/sharing/webmasters/crawler)"
+    - ~destination_address: >-
+         <destination.ip>(?:<destination.port>), <~ip1/ip>(?,) <~ip2/ip>(?,) <~ip3/ip>
+
+    # 3 Ips
+    # 10.0.0.2, 10.0.0.1, 127.0.0.1 - - [07/Dec/2016:11:05:07 +0100] "GET /ocelot HTTP/1.1" 200 571 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:49.0) Gecko/20100101 Firefox/49.0"
+    # 2a02:cf40:10ff:f00f:0000:0000:0:8000, 10.225.192.17 10.2.2.121 - - [30/Dec/2016:06:47:09 +0000] "GET /test.html HTTP/1.1" 404 8571 "-" "Mozilla/5.0 (compatible; Facebot 1.0; https://developers.facebook.com/docs/sharing/webmasters/crawler)"
+    - ~destination_address: >-
+         <destination.ip>(?:<destination.port>), <~ip1/ip>(?,) <~ip2/ip>
+
+    # Domain + 3 IPs
+    # example.com 10.0.0.2, 10.0.0.1, 81.2.69.143 - - [07/Dec/2016:11:05:07 +0100] "GET /ocelot HTTP/1.1" 200 571 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:49.0) Gecko/20100101 Firefox/49.0"
+    - ~destination_address: >-
+        <~destination_domain/fqdn>(?:<destination.port>) <~ip1/ip>(?,) <~ip2/ip>(?,) <~ip3/ip>
+
+    # 3 domains
+    # example.com localhost, localhost - - [29/May/2017:19:02:48 +0000] "GET /test2 HTTP/1.1" 200 612 "-" "Mozilla/5.0 (Windows NT 6.1; rv:15.0) Gecko/20120716 Firefox/15.0a2" "-"
+    - ~destination_address: >-
+        <~destination_domain/fqdn>(?:<destination.port>) <~domain1/fqdn>(?,) <~domain2/fqdn>
+
+    # 2 Ips
+    # 1.128.3.4:80 127.0.0.1 - - [12/Apr/2018:09:48:40 +0200] "" 400 0 "-" "-"
+    - ~destination_address: <?destination.ip>(?:<destination.port>) <~source_address/ip>
+
+    # Domain + IP
+    # lessons.example.com 192.168.0.1 - - [09/Jun/2020:12:10:39 -0700] "GET /A%20Beka%20G1%20Howe/029_AND_30/15%20reading%20elephants.mp4 HTTP/1.1" 206 7648063 "http://lessons.example.com/A%20Beka%20G1%20Howe/029_AND_30/15%20reading%20elephants.mp4" "Mozilla/5.0 (Linux; Android 5.1.1; KFFOWI) AppleWebKit/537.36 (KHTML, like Gecko) Silk/81.2.16 like Chrome/81.0.4044.138 Safari/537.36"
+    - ~destination_address: <~destination_domain/fqdn>(?:<destination.port>) <~source_address/ip>
+
+    # IP + domain
+    # 1.128.3.4 localhost - - [29/May/2017:19:02:48 +0000] "GET /test2 HTTP/1.1" 200 612 "-" "Mozilla/5.0 (Windows NT 6.1; rv:15.0) Gecko/20120716 Firefox/15.0a2" "-"
+    - ~destination_address: <destination.ip>(?:<destination.port>) <~domain1/fqdn>
```

Comment:
What is the difference? An ip is also an FQDN. You can map the fqdn and if possible then get the ip with the ip parser in the normalize step.

Reply:
It replicates the rest of the decoder

Reply:
I tried to unify it that way, it would be very helpful but I found that when using IPv6 addresses and having the optional port, there's an issue confusing the port with part of the address:
```

[decoder/nginx-access/0] [/~0destination_address: <~destination_domain/fqdn>(?:<destination.port>), <~ip1/ip>(?,) <~ip2/ip>] -> Failure: Parse operation failed: 
Main error: <destination.port>: Expected a number at 5
2a02:cf40:10ff:f00f:0000:0000:0:8000, 10.225.192.17 10.2.2.121
-----^

List of errors:
<destination.port>: Expected a number at 5
<~destination_domain/fqdn>: Invalid char ':' found while parsing at 0

```